src/body/Body.js

```diff
@@ -503,45 +524,116 @@ var Axes = require('../geometry/Axes');
      * @param {vector} velocity
      */
     Body.setVelocity = function(body, velocity) {
-        body.positionPrev.x = body.position.x - velocity.x;
-        body.positionPrev.y = body.position.y - velocity.y;
-        body.velocity.x = velocity.x;
-        body.velocity.y = velocity.y;
+        var timeScale = body.deltaTime / Common._timeUnit;
+        body.positionPrev.x = body.position.x - velocity.x * timeScale;
+        body.positionPrev.y = body.position.y - velocity.y * timeScale;
+        body.velocity.x = velocity.x * timeScale;
+        body.velocity.y = velocity.y * timeScale;
```

Comment:
The changes you made to the way velocity work broke something in my game. It's a game where you have a character that can shoot bodies while moving. I want the speed of the bullet to be the addition of the speed of the character and a force at which the bullet is thrown. I was doing this by:

- making a bullet body
- setting the velocity of the bullet to the velocity of the character
- applying a force to "throw" the bullet

Obviously, as [you warned here](https://github.com/liabru/matter-js/blob/82bb41535b09e02dd76a20d5955fc7b4102487d6/src/body/Body.js#L881), I shouldn't have done it this way, so it's okay that it broke (I'm doing it properly now :stuck_out_tongue: ). But I thought you might want to know. :slightly_smiling_face: